src/Parser/BalanceDruid/Modules/Features/AstralPower.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import Icon from 'common/Icon';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import Module from 'Parser/Core/Module';
+import SPELLS from 'common/SPELLS';
+
+class AstralPower extends Module {
+  
+  lastAstral = 0;
+  aspWasted = 0;
+
+  on_toPlayer_energize(event) {
+      for (let i = 0; i < event.classResources.length; i++) {
+          if (event.classResources[i].type === 8){
```

Comment:
Please extract magic numbers to make things easier to understand. You can find a resources types object in `common/RESOURCE_TYPES` (this is `ASTRAL_POWER`).

Reply:
Repeated below